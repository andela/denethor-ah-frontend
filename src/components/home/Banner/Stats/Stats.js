import React from 'react';
import './styles.scss';

const defaultProps = {
  writers: 100000,
  readers: 344555,
  topAuthors: [{
    imageUrl: '/assets/images/author1.png',
    firstName: 'Omoefe',
    articles: 674,
    followers: 783
  }, {
    imageUrl: '/assets/images/author2.png',
    firstName: 'Ibidapo',
    articles: 60,
    followers: 45
  }]
};

const Stats = (props) => {
  if(!props.writers) {
    props = defaultProps;
  }

  const { writers, readers, topAuthors: [author1, author2] } = props;

  const topAuthor = ({imageUrl, firstName, articles, followers }) => (
    <div className="author">
      <img src={imageUrl} alt={firstName} />
      <div>
        <h3>{firstName}</h3>
        <p><span className="banner__yellow-text">{`${articles}+`}</span> articles</p>
        <p><span className="banner__yellow-text">{`${followers}+`}</span> followers</p>
      </div>
    </div>
  );

  return (
      <div className="stats">
        <div className="stats__numbers stats__numbers--writers">
          <h2 className="banner__off-white-text">Connect with</h2>
          <h2>
            <span className="banner__yellow-text">{`${writers}+ `}</span>
            <span className="banner__off-white-text">writers</span>
          </h2>
        </div>
        <div className="stats__top-authors">
          <div>
            <h1 className="hidden-on-mobile">Join these <br /> successful writers</h1>
            <div className="stats__top-authors__authors">
              {topAuthor(author1)}
              {topAuthor(author2)}
            </div>
          </div>
        </div>
        <div className="stats__numbers stats__numbers--readers">
          <h2 className="banner__off-white-text">Publish to</h2>
          <h2>
            <span className="banner__yellow-text">{`${readers}+ `} </span>
            <span className="banner__off-white-text">readers</span>
          </h2>
        </div>
      </div>
  );
};

export default Stats;