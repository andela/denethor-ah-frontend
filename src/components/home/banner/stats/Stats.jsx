import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import './styles.scss';


export const Stats = (props) => {

  const { writers, readers, topAuthors: [author1, author2] } = props;

  const topAuthor = ({ imageUrl, firstname, articlesWritten, followers }) => (
    <div className="author">
      <img src={imageUrl} alt={firstname} />
      <div>
        <h3>{firstname}</h3>
        <p><span className="banner__yellow-text">{`${articlesWritten}+`}</span> articles</p>
        <p><span className="banner__yellow-text">{`${followers}+`}</span> followers</p>
      </div>
    </div>
  );

  topAuthor.propTypes={
    imageUrl: PropTypes.string,
    firstname: PropTypes.string,
    articlesWritten: PropTypes.number,
    followers: PropTypes.number
  }

  return (
    <div className="stats">
      <div className="stats__numbers stats__numbers--writers">
        <div>
          <h2 className="banner__off-white-text">Connect with</h2>
          <h2>
            <span className="banner__yellow-text">{`${writers}+ `}</span>
            <span className="banner__off-white-text">writers</span>
          </h2>
        </div>
      </div>
      <div className="stats__top-authors">
        <div>
          <h1 className="hidden-on-mobile">Join these <br /> successful writers</h1>
          <div className="stats__top-authors__authors">
            {topAuthor(author1)}
            {topAuthor(author2)}
          </div>
        </div>
      </div>
      <div className="stats__numbers stats__numbers--readers">
        <div>
          <h2 className="banner__off-white-text">Publish to</h2>
          <h2>
            <span className="banner__yellow-text">{`${readers}+ `} </span>
            <span className="banner__off-white-text">readers</span>
          </h2>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = ({ users }) => ({
  writers: 12545,
  readers: 63843456,
  topAuthors: users.filter(({ id }) => (id === 'top1' || id === 'top2'))
});

Stats.propTypes={
 writers: PropTypes.number,
 readers: PropTypes.number,
 topAuthors: PropTypes.array
}

export default connect(mapStateToProps)(Stats);
