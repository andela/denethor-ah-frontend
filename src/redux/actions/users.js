import actions from './types';


export const addUser = user => (dispatch) => {
  dispatch({
    type: actions.ADD_USER,
    user
  });
};

export const removeUser = id => (dispatch) => {
  dispatch({
    type: actions.REMOVE_USER,
    id
  });
};
