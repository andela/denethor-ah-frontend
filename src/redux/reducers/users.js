import { ADD_USER, REMOVE_USER } from '../actions/types';

const userReducerDefaultState = [{
  "id": "top1",
  "firstname": "Muna",
  "lastname": "Dukuye",
  "username": "omoisgood7179",
  "role": "author",
  "notifications": true,
  "bio": null,
  "imageUrl": "https://i.imgur.com/d4AI2Nh.jpg",
  "createdAt": "2019-03-07T21:05:08.071Z",
  "followers": 432485,
  "userArticles": [{
    "id": "8ee5e8ed-ecdf-41c1-9b94-6c6bb712a77a",
    "slug": "they-said-the-titile-length-must-be-at-least-5",
    "description": "description is not allowed to be empty so I added this"
  }, {
    "id": "8ee5e8ed-ecdf-41c1-9b94-6c6bb712a77a",
    "slug": "they-said-the-titile-length-must-be-at-least-5",
    "description": "description is not allowed to be empty so I added this"
  }],
  "articlesWritten": 193
}, {
  "id": "top2",
  "firstname": "Omoefe",
  "lastname": "Dukuye",
  "username": "omoisgood7179",
  "role": "author",
  "notifications": true,
  "bio": null,
  "imageUrl": "https://i.imgur.com/ws9TUUN.jpg",
  "createdAt": "2019-03-07T21:05:08.071Z",
  "followers": 232848,
  "userArticles": [{
    "id": "8ee5e8ed-ecdf-41c1-9b94-6c6bb712a77a",
    "slug": "they-said-the-titile-length-must-be-at-least-5",
    "description": "description is not allowed to be empty so I added this"
  }],
  "articlesWritten": 126
}];

export default (state = userReducerDefaultState, action) => {
  switch (action.type) {
    case ADD_USER:
      return [...state, action.user];

    case REMOVE_USER:
      return state.filter(({ id }) => id !== action.id );

    default:
      return state;
  }
};