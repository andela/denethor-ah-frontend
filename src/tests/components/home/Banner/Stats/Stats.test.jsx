import React from 'react';
import { shallow } from 'enzyme';
import { Stats } from '../../../../../components/home/Banner/Stats/Stats';
import users from '../../../../mock-data/users'

it('Should render stats component correctly', () => {
  const props = {
    topAuthors: [users[0], users[1]]
  }
  const wrapper = shallow(<Stats {...props}/>);
  expect(wrapper).toMatchSnapshot();

});
