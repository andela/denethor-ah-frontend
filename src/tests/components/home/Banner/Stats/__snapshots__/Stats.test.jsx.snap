// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render stats component correctly 1`] = `
<div
  className="stats"
>
  <div
    className="stats__numbers stats__numbers--writers"
  >
    <div>
      <h2
        className="banner__off-white-text"
      >
        Connect with
      </h2>
      <h2>
        <span
          className="banner__yellow-text"
        >
          undefined+ 
        </span>
        <span
          className="banner__off-white-text"
        >
          writers
        </span>
      </h2>
    </div>
  </div>
  <div
    className="stats__top-authors"
  >
    <div>
      <h1
        className="hidden-on-mobile"
      >
        Join these 
        <br />
         successful writers
      </h1>
      <div
        className="stats__top-authors__authors"
      >
        <div
          className="author"
        >
          <img
            alt="Omo"
            src="the-first-image"
          />
          <div>
            <h3>
              Omo
            </h3>
            <p>
              <span
                className="banner__yellow-text"
              >
                12+
              </span>
               articles
            </p>
            <p>
              <span
                className="banner__yellow-text"
              >
                21+
              </span>
               followers
            </p>
          </div>
        </div>
        <div
          className="author"
        >
          <img
            alt="Omoefe"
            src="the-second-image"
          />
          <div>
            <h3>
              Omoefe
            </h3>
            <p>
              <span
                className="banner__yellow-text"
              >
                20+
              </span>
               articles
            </p>
            <p>
              <span
                className="banner__yellow-text"
              >
                78+
              </span>
               followers
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="stats__numbers stats__numbers--readers"
  >
    <div>
      <h2
        className="banner__off-white-text"
      >
        Publish to
      </h2>
      <h2>
        <span
          className="banner__yellow-text"
        >
          undefined+ 
           
        </span>
        <span
          className="banner__off-white-text"
        >
          readers
        </span>
      </h2>
    </div>
  </div>
</div>
`;
